### PR TITLE
Form

### PR DESCRIPTION
**Pull request by will-gant**
This pull request creates a feature that adds an image uploader to the 'new posts' view. All submitted images are saved by carrierwave (a gem) to public/uploads/post/image.

The same gem creates an 'image' column in the posts table, which points to the location of the uploaded images on the server.

We have also changed the posts/index.html.erb page to display images whenever these are associated with a post.

### CODE REVIEW

**Comment by BertZZ** [comment]
@bruxelles86 Your Travis failure may have something to do with the new routing logic @TimRobinson1 recently pushed to master. Sign In a user in your tests first, and fix your rubocop

**Review by BertZZ — COMMENTED** [review]
Ran Tests and all pass. Feature works on localhost. No obvious errors. Send it to @neoeno

File: Gemfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ git_source(:github) do |repo_name|
   "https://github.com/#{repo_name}.git"
 end
 
+gem "carrierwave"
```
Comment (neoeno) [inline Gemfile]: I'd probably shove this gem further down (the gemfile acts as a sort of de-facto documentation, so we want to roughly have the most important stuff first) and comment it to say what it's used for, like the other gems here are.

Usually our code tells us what it's doing — but dependencies have weird names so it's good to leave a comment here just to explain what it does and why we're using it!

File: spec/features/user_posts_photos_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.feature 'Users can post photos to the timeline', type: :feature do
+  scenario 'Uploaded images appear on the next screen' do
+    sign_in
+    click_link 'New post'
+    attach_file('post_image', Rails.root + 'public/owl.jpg')
+    fill_in 'Message', with: 'Test post'
+    click_button 'Submit'
+    expect(page).to have_xpath("//img[contains(@src,'owl.jpg')]")
```
Comment (neoeno) [inline spec/features/user_posts_photos_spec.rb]: CSS selectors might be more common than xpath selectors here —

```ruby
expect(page).to have_css("img[src*='owl.jpg']")
```

In this case I think the XPath is more readable, so I'd leave it in, but be aware CSS selectors are usually preferred.

File: spec/features/user_posts_photos_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.feature 'Users can post photos to the timeline', type: :feature do
+  scenario 'Uploaded images appear on the next screen' do
+    sign_in
+    click_link 'New post'
+    attach_file('post_image', Rails.root + 'public/owl.jpg')
```
Comment (neoeno) [inline spec/features/user_posts_photos_spec.rb]: Usually these sorts of test image files, etc are put in `spec/support/assets`

**Review by neoeno — CHANGES_REQUESTED** [review]
Great! Just a few comments, mostly about conventions. I'd suggest watching for the whitespace changes you're introducing — not sure if all of them were intended.

**Comment by ffasolin** [comment]
@neoeno Suggested changes have been implemented

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1,8 +1,7 @@
 <% @posts.order(created_at: :desc).each do |post| %>
-
   <%= simple_format(post.message) %> <p>Posted on <%= post.created_at.strftime("%d/%m/%Y")%> at <%= post.created_at.strftime("%H:%M")%></p>
   <%= image_tag post.image_url.to_s %>
-
+  <%= post_presenter(post) %>
```
Comment (neoeno) [inline app/views/posts/index.html.erb]: This merge isn't supposed to be like this I think — open it up in the browser.

**Review by neoeno — CHANGES_REQUESTED** [review]
There's a secret spider dog lurking around in the `public` directory still.

**Review by neoeno — APPROVED** [review]
Looks like just one rubocop fix and you're good.

**Pull request merged**